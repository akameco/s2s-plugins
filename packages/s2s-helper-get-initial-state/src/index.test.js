// @flow
const helper = require('./').default

test('get initalState', () => {
  const fixture = `
export const initialState = {count: 1}
`

  const actions = helper(fixture)
  expect(actions).toMatchSnapshot()
})

test('get initalState with flow', () => {
  const fixture = `
export const initialState: State = {count: 1}
`

  const actions = helper(fixture)
  expect(actions).toMatchSnapshot()
})

test('not get initalState', () => {
  const fixture = `
export const x = {count: 1}
`

  const actions = helper(fixture)
  expect(actions).toBe(null)
})
