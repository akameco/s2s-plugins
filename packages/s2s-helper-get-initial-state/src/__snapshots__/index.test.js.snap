// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get initalState 1`] = `
Node {
  "end": 39,
  "loc": SourceLocation {
    "end": Position {
      "column": 38,
      "line": 2,
    },
    "start": Position {
      "column": 28,
      "line": 2,
    },
  },
  "properties": Array [
    Node {
      "computed": false,
      "end": 38,
      "key": Node {
        "end": 35,
        "loc": SourceLocation {
          "end": Position {
            "column": 34,
            "line": 2,
          },
          "identifierName": "count",
          "start": Position {
            "column": 29,
            "line": 2,
          },
        },
        "name": "count",
        "start": 30,
        "type": "Identifier",
      },
      "loc": SourceLocation {
        "end": Position {
          "column": 37,
          "line": 2,
        },
        "start": Position {
          "column": 29,
          "line": 2,
        },
      },
      "method": false,
      "shorthand": false,
      "start": 30,
      "type": "ObjectProperty",
      "value": Node {
        "end": 38,
        "extra": Object {
          "raw": "1",
          "rawValue": 1,
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 37,
            "line": 2,
          },
          "start": Position {
            "column": 36,
            "line": 2,
          },
        },
        "start": 37,
        "type": "NumericLiteral",
        "value": 1,
      },
    },
  ],
  "start": 29,
  "type": "ObjectExpression",
}
`;

exports[`get initalState with flow 1`] = `
Node {
  "end": 46,
  "loc": SourceLocation {
    "end": Position {
      "column": 45,
      "line": 2,
    },
    "start": Position {
      "column": 35,
      "line": 2,
    },
  },
  "properties": Array [
    Node {
      "computed": false,
      "end": 45,
      "key": Node {
        "end": 42,
        "loc": SourceLocation {
          "end": Position {
            "column": 41,
            "line": 2,
          },
          "identifierName": "count",
          "start": Position {
            "column": 36,
            "line": 2,
          },
        },
        "name": "count",
        "start": 37,
        "type": "Identifier",
      },
      "loc": SourceLocation {
        "end": Position {
          "column": 44,
          "line": 2,
        },
        "start": Position {
          "column": 36,
          "line": 2,
        },
      },
      "method": false,
      "shorthand": false,
      "start": 37,
      "type": "ObjectProperty",
      "value": Node {
        "end": 45,
        "extra": Object {
          "raw": "1",
          "rawValue": 1,
        },
        "loc": SourceLocation {
          "end": Position {
            "column": 44,
            "line": 2,
          },
          "start": Position {
            "column": 43,
            "line": 2,
          },
        },
        "start": 44,
        "type": "NumericLiteral",
        "value": 1,
      },
    },
  ],
  "start": 36,
  "type": "ObjectExpression",
}
`;
